File: lab 5/c++/main.cpp
#include <iostream>
#include <string>
#include <windows.h>
#include "String.h"
using namespace std;

void ShowInfo(Strings* str) {
    SetConsoleOutputCP(65001);
    std::cout << "\nОригинальний рядок: " << str->getString() << std::endl;
    std::cout << "Без заборонених символів: " << str->withoutChar() << std::endl;
    std::cout << "Довжина фінального рядка: " << str->stringLength() << std::endl;
}

int main() {
    SetConsoleOutputCP(65001);
    string a, b;

    cout << "Введіть перший рядок: ";
    getline(cin, a);

    cout << "Введіть другий рядок: ";
    getline(cin, b);

    Strings* n = new Numbers(a);
    Strings* l = new Letters(b);

    ShowInfo(n);
    ShowInfo(l);

    delete n;
    delete l;

    return 0;
}

File: lab 5/c#/Main.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp9
{
    class Program
    {
        static void Main()
        {
            Console.Write("Введіть пешрий рядок: ");
            string a = Console.ReadLine();

            Console.Write("Введіть другий рядок: ");
            string b = Console.ReadLine();

            Strings n = new Numbers(a);
            Strings l = new Letters(b);
            ShowInfo(n); 
            ShowInfo(l); 
        }

        static void ShowInfo(Strings str)
        {
            Console.WriteLine();
            Console.WriteLine("Оригинальний рядок: " + str.getString());
            Console.WriteLine("Без заборонених символів: " + str.withoutChar());
            Console.WriteLine("Довжина рядка: " + str.stringLength());
        }
    }
}
